src/sentry/shared_integrations/client/proxy.py

```diff
@@ -23,30 +26,54 @@ class IntegrationProxyClient(ApiClient):  # type: ignore
     Universal Client to access third-party resources safely in Hybrid Cloud.
     Requests to third parties must always exit the Sentry subnet via the Control Silo, and only
     add sensitive credentials at that stage.
+
+    When testing, client requests will always go to the base_url unless `self.use_proxy_url_for_tests`
+    is set to True.
     """
 
     should_proxy_to_control = False
+    use_proxy_url_for_tests = False
 
-    def __init__(self, org_integration_id: int) -> None:
+    def __init__(
+        self,
+        integration_id: int | None = None,
+        org_integration_id: int | None = None,
+    ) -> None:
         super().__init__()
+
+        self.integration_id = integration_id
         self.org_integration_id = org_integration_id
 
         is_region_silo = SiloMode.get_current_mode() == SiloMode.REGION
-        subnet_secret = getattr(settings, "SENTRY_SUBNET_SECRET")
-        control_address = getattr(settings, "SENTRY_CONTROL_ADDRESS")
+        subnet_secret = getattr(settings, "SENTRY_SUBNET_SECRET", None)
+        control_address = getattr(settings, "SENTRY_CONTROL_ADDRESS", None)
+        is_test_environment = "pytest" in sys.modules
 
         if is_region_silo and subnet_secret and control_address:
             self.should_proxy_to_control = True
             self.proxy_url = f"{settings.SENTRY_CONTROL_ADDRESS}{PROXY_BASE_PATH}"
 
+        if is_test_environment and not self.use_proxy_url_for_tests:
+            self.proxy_url = self.base_url
```

Comment:
This is the work around for avoiding having to modify every `responses.add('slack.com')` call across the two code bases. The only ones that should fail now are tests that explicitly check for credentials in the region silo, which are expected now.

If someone did want to ensure the request was proxied, they could also check for the subnet signature headers.